examples/rust/src/main.rs

```diff
@@ -1,198 +1,39 @@
-        Err((rejection_code, message)) => trap(&format!(
-            "Received a reject from Bitcoin canister.\nRejection Code: {:?}\nMessage: '{}'",
-            rejection_code, message
-        )),
-    }
-}
-
-/// Returns the canister's balance.
-#[update]
-pub async fn balance() -> u64 {
-    let btc_canister_id = BTC_CANISTER_ID.with(|id| *id.borrow());
-    let res: Result<(Result<u64, GetBalanceError>,), (RejectionCode, String)> = call(
-        btc_canister_id,
-        "get_balance",
-        (GetBalanceRequest {
-            address: btc_address_str(),
-            min_confirmations: Some(0),
-        },),
-    )
-    .await;
-
-    match res {
-        // Return the balance to the caller.
-        Ok((Ok(balance),)) => balance,
-
-        // The call to `get_balance` returned an error.
-        Ok((Err(err),)) => trap(&format!("Received error from Bitcoin canister: {:?}", err)),
-
-        // The call to `get_balance` was rejected.
-        // This is only likely to happen if there's a bug in the bitcoin canister.
-        Err((rejection_code, message)) => trap(&format!(
-            "Received a reject from Bitcoin canister.\nRejection Code: {:?}\nMessage: '{}'",
-            rejection_code, message
-        )),
-    }
-}
-
-/// Send the `amount` of satoshis provided to the `destination` address.
-///
-/// Notes:
-///  * Fees are hard-coded to 10k satoshis.
-///  * Input UTXOs are not being selected in any smart way.
-///  * A dust threshold of 10k satoshis is used.
 #[update]
-pub async fn send(amount: u64, destination: String) {
-    let fees: u64 = 10_000;
-
-    if amount <= fees {
-        trap("Amount must be higher than the fee of 10,000 satoshis")
-    }
-
-    let destination = match Address::from_str(&destination) {
-        Ok(destination) => destination,
-        Err(_) => trap("Invalid destination address"),
-    };
-
-    // Fetch our UTXOs.
-    let utxos = get_utxos().await;
-
-    // Remove any spent UTXOs that were already used for past transactions.
-    let utxos = utxos
-        .into_iter()
-        .filter(|utxo| SPENT_TXOS.with(|spent_txos| !spent_txos.borrow().contains(&utxo.outpoint)))
-        .collect();
-
-    let spending_transaction = build_transaction(utxos, btc_address(), destination, amount, fees)
-        .unwrap_or_else(|err| {
-            trap(&format!("Error building transaction: {}", err));
-        });
-
-    // Cache the spent outputs to not use them for future transactions.
-    for tx_in in spending_transaction.input.iter() {
-        SPENT_TXOS.with(|spent_txos| {
-            print(&format!("Caching {:?}", tx_in.previous_output.txid.to_vec()));
-            spent_txos.borrow_mut().insert(OutPoint {
-                txid: tx_in.previous_output.txid.to_vec(),
-                vout: tx_in.previous_output.vout,
-            })
-        });
+pub async fn test() -> (String, Vec<Utxo>, u64, String, String) {
+    let recipient = String::from("bcrt1qyepqdteh7w9fhtsrylzghduhj4th5260ae3ywf");
```

Comment:
What's this string? A Bitcoin address starts with 1, 3, or bc1q (or bc1p for taproot).